service/orchestrator/assessment_results.go

```diff
@@ -47,8 +48,8 @@ func (svc *Service) ListAssessmentResults(ctx context.Context, req *assessment.L
 	// only allowed, if one can access all the cloud services. Furthermore, the content of the filtered cloud service ID
 	// must be in the list of allowed cloud service IDs.
 	all, allowed = svc.authz.AllowedCloudServices(ctx)
-	if (req.FilteredCloudServiceId == "" && !all) ||
-		(req.FilteredCloudServiceId != "" && !slices.Contains(allowed, req.FilteredCloudServiceId)) {
+	if !all && (req.FilteredCloudServiceId == "" ||
```

Comment:
Should be solid now. Could we actually add more tests to this since we obviously did not catch this mistake with our current tests?